controller/service.go

```diff
@@ -30,14 +30,40 @@ import (
 )
 
 const (
-	AnnotationPrefix             = "metallb.universe.tf"
+	AnnotationPrefix             = "metallb.io"
 	AnnotationAddressPool        = AnnotationPrefix + "/" + "address-pool"
 	AnnotationLoadBalancerIPs    = AnnotationPrefix + "/" + "loadBalancerIPs"
 	AnnotationIPAllocateFromPool = AnnotationPrefix + "/" + "ip-allocated-from-pool"
+	AnnotationAllowSharedIP      = AnnotationPrefix + "/" + "allow-shared-ip"
+
+	// Deprecated Annotations. Used for backward compatibility.
+	DeprecatedAnnotationPrefix             = "metallb.universe.tf"
+	DeprecatedAnnotationAddressPool        = DeprecatedAnnotationPrefix + "/" + "address-pool"
+	DeprecatedAnnotationLoadBalancerIPs    = DeprecatedAnnotationPrefix + "/" + "loadBalancerIPs"
+	DeprecatedAnnotationIPAllocateFromPool = DeprecatedAnnotationPrefix + "/" + "ip-allocated-from-pool"
+	DeprecatedAnnotationAllowSharedIP      = DeprecatedAnnotationPrefix + "/" + "allow-shared-ip"
 )
 
 var ErrConverge = fmt.Errorf("failed to converge")
 
+// SharingKey extracts the sharing key for a service.
+func SharingKey(svc *v1.Service) string {
+	if _, ok := svc.Annotations[AnnotationAllowSharedIP]; ok {
+		return svc.Annotations[AnnotationAllowSharedIP]
+	}
+	return svc.Annotations[DeprecatedAnnotationAllowSharedIP]
+}
+
+func getAnnotationValue(str ...string) string {
```

Comment:
I'd:
- move this to the bottom as this is an "utility-like" function
- rework this by providing the annotations map, the supported annotation and the deprecated annotation:

`func valueForAnnotation(annotations map[string]string, stableAnnotation, deprecatedAnnotation) string` as its implementation will be more explicit than fetching the first one and relying on the caller setting the first as the stable one

Reply:
I changed the implementation based on your input and moved the function to the bottom of the file as you requested.